kernel/src/expressions/scalars.rs

```diff
@@ -85,7 +85,7 @@ impl Display for Scalar {
                 }
                 Ordering::Less => {
                     write!(f, "{}", value)?;
-                    for _ in 0..(scale.abs()) {
+                    for _ in 0..(*scale) {
```

Comment:
nit: do we actually need the `()` for this? I would have expected operator precedence to take care of it?
```rust
for _ in 0 .. *scale {
```